### PR TITLE
Sqlalchemy 2.0 changes

### PR DESCRIPTION
**Pull request by phanikumv**
Part of https://github.com/apache/airflow/issues/28723

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/trigger_dagrun.py [hunk 1]
```diff
@@ -227,12 +228,10 @@ def execute_complete(self, context: Context, session: Session, event: tuple[str,
...
         provided_execution_date = event[1]["execution_dates"][0]
         try:
-            dag_run = (
-                session.query(DagRun)
-                .filter(
+            dag_run = session.scalar(
+                select(DagRun).where(
                     DagRun.dag_id == self.trigger_dag_id, DagRun.execution_date == provided_execution_date
                 )
-                .one()
```
Comment (Taragolis) [inline airflow/operators/trigger_dagrun.py]: If we rewrite by this way then `NoResultFound` would never raised

I assume we should use either use `session.scalars(...).one()` ([ScalarResult](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.ScalarResult) methods) or check that `dag_run is not None`

Comment (uranusjr) [reply airflow/operators/trigger_dagrun.py]: I’ve pushed a commit directly to this branch to fix this.

**Review by Taragolis — APPROVED** [review]

File: airflow/operators/trigger_dagrun.py (hunk 1 above)
Comment (phanikumv) [reply airflow/operators/trigger_dagrun.py]: sqlite tests are failing after this commit

**Review by uranusjr — APPROVED** [review]

File: airflow/operators/trigger_dagrun.py (hunk 1 above)
Comment (phanikumv) [reply airflow/operators/trigger_dagrun.py]: test failures were transient, re run went through fine

**Pull request merged**